### PR TITLE
Replace Cast with NumericCast

### PR DESCRIPTION
**Pull request by Nugine**
<https://docs.rs/numeric_cast>

### CODE REVIEW

**Comment by pzheng1025** [comment]
What is the benefit of using numeric_cast comparing to the old implementation?

**Comment by Nugine** [comment]
> What is the benefit of using numeric_cast comparing to the old implementation?

Described in the docs
https://docs.rs/numeric_cast

File: src/conversion.rs [hunk 1]
```diff
@@ -1,29 +1,6 @@
...
-                type_name::<Self>(),
-                self,
-                type_name::<T>(),
-            )
-        })
-    }
-}
-
-impl<U> Cast for U {}
+pub use numeric_cast::NumericCast;
```
Comment (rogercloud) [inline src/conversion.rs:3]: It limits other types casting.

Comment (Nugine) [reply src/conversion.rs:3]: Other types can use `.try_into().expect(...)`, `.try_into().unwrap()`, `.try_into().unwrap_or_else(...)`. 
They don't need the trait `Cast`.

Comment (pzheng1025) [reply src/conversion.rs:3]: The initial purpose of cast is to reduce the verbose code like try_into().unwrap_or_else().

Comment (pzheng1025) [reply src/conversion.rs:3]: Is it possible that for num using numeric_cast::NumericCast and others use cast?

Comment (Nugine) [reply src/conversion.rs:3]: 99% use cases are number conversions. So I think the replacement is ok.

Comment (Nugine) [reply src/conversion.rs:3]: > Is it possible that for num using numeric_cast::NumericCast and others use cast?

It needs a nightly feature: specialization.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**